proposals/lip-0055.md

```diff
@@ -300,12 +342,130 @@ blockHeaderSchema = {
     },
     "signature": {
       "dataType": "bytes",
-      "fieldNumber": 13
+      "fieldNumber": 14
+    }
+  }
+}
+```
+
+#### Unsigned Block Header JSON Schema
+
+Unsigned block headers are serialized and deserialized accordingly to the following JSON schema (notice the missing `signature` property).
+
+```java
+unsignedBlockHeaderSchema = {
+  "type": "object",
+  "required": [
+    "version",
+    "timestamp",
+    "height",
+    "previousBlockID",
+    "generatorAddress",
+    "transactionRoot",
+    "assetRoot",
+    "eventRoot",
+    "stateRoot",
+    "maxHeightPrevoted",
+    "maxHeightGenerated",
+    "validatorsHash",
+    "aggregateCommit"
+  ],
+  "properties": {
+    "version": {
+      "dataType": "uint32",
+      "fieldNumber": 1
+    },
+    "timestamp": {
+      "dataType": "uint32",
+      "fieldNumber": 2
+    },
+    "height": {
+      "dataType": "uint32",
+      "fieldNumber": 3
+    },
+    "previousBlockID": {
+      "dataType": "bytes",
+      "fieldNumber": 4
+    },
+    "generatorAddress": {
+      "dataType": "bytes",
+      "fieldNumber": 5
+    },
+    "transactionRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 6
+    },
+    "assetRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 7
+    },
+    "eventRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 8
+    },
+    "stateRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 9
+    },
+    "maxHeightPrevoted": {
+      "dataType": "uint32",
+      "fieldNumber": 10
+    },
+    "maxHeightGenerated": {
+      "dataType": "uint32",
+      "fieldNumber": 11
+    },
+    "validatorsHash": {
+      "dataType": "bytes",
+      "fieldNumber": 12
+    },
+    "aggregateCommit": {
+      "type": "object",
+      "fieldNumber": 13,
+      "required": [
+        "height",
+        "aggregationBits",
+        "certificateSignature"
+      ],
+      "properties": {
+        "height": {
+          "dataType": "uint32",
+          "fieldNumber": 1
+        },
+        "aggregationBits": {
+          "dataType": "bytes",
+          "fieldNumber": 2
+        },
+        "certificateSignature": {
+          "dataType": "bytes",
+          "fieldNumber": 3
+        }
+      }
     }
   }
 }
 ```
 
+
+#### Block Signature Calculation
+
+The block header signature is calculated using the `calculateBlockSignature` function. This function returns a bytes value with length `SIGNATURE_LENGTH_BYTES`.
```

Comment:
Maybe we could rename `SIGNATURE_LENGTH_BYTES` to `SIGNATURE_LENGTH_EDDSA_BYTES` as we also have BLS signatures which have a different byte size.